pytools/graph.py

```diff
@@ -47,6 +48,7 @@
 .. autofunction:: as_graphviz_dot
 .. autofunction:: validate_graph
 .. autofunction:: is_connected
+.. autofunction:: get_propagation_graph_from_constraints
```

Comment:
This is still missing `get_reachable_nodes` as well as a (probably longer) narrative on what these functions are even *for*.